fluss-server/src/main/java/com/alibaba/fluss/server/kv/snapshot/PeriodicSnapshotManager.java

```diff
@@ -177,34 +177,36 @@ public void triggerSnapshot() {
         // of using guardedExecutor
         guardedExecutor.execute(
                 () -> {
-                    LOG.debug("TableBucket {} triggers snapshot.", tableBucket);
-                    long triggerTime = System.currentTimeMillis();
-
-                    Optional<SnapshotRunnable> snapshotRunnableOptional;
-                    try {
-                        snapshotRunnableOptional = target.initSnapshot();
-                    } catch (Exception e) {
-                        LOG.error("Fail to init snapshot during triggering snapshot.", e);
-                        return;
-                    }
-                    if (snapshotRunnableOptional.isPresent()) {
-                        SnapshotRunnable runnable = snapshotRunnableOptional.get();
-                        asyncOperationsThreadPool.execute(
-                                () ->
-                                        asyncSnapshotPhase(
-                                                triggerTime,
-                                                runnable.getSnapshotId(),
-                                                runnable.getCoordinatorEpoch(),
-                                                runnable.getBucketLeaderEpoch(),
-                                                runnable.getSnapshotLocation(),
-                                                runnable.getSnapshotRunnable()));
-                    } else {
-                        scheduleNextSnapshot();
-                        LOG.debug(
-                                "TableBucket {} has no data updates since last snapshot, "
-                                        + "skip this one and schedule the next one in {} seconds",
-                                tableBucket,
-                                periodicSnapshotDelay / 1000);
+                    if (started && !periodicExecutor.isShutdown()) {
```

Comment:
Why add `if (started && !periodicExecutor.isShutdown())` in here?

Reply:
For periodicExecutor summitting snapshot trigger with delay，after bucket deleted and `started` change to false, the snapshot trigger will execute again for last time and generate an empty snapshot, the remote bucket dir will be created again，so i add this code for avoiding this case.

Reply:
Thanks for the explanation...Make sense to me..But we won't need to check `!periodicExecutor.isShutdown()`, it `periodicExecutor.isShutdown()`, the task won't be scheduled..